h2o-py/tests/testdir_algos/gbm/pyunit_PUBDEV_7578_gbm_reproducibility.py

```diff
@@ -0,0 +1,240 @@
+        learn_rate_annealing = 1.0,
+        distribution = 'bernoulli',
+        quantile_alpha = 0.5,
+        tweedie_power = 1.5,
+        huber_alpha = 0.9,
+        sample_rate = 0.75,
+        col_sample_rate = 0.31,
+        col_sample_rate_change_per_level = 0.94,
+        col_sample_rate_per_tree = 0.65,
+        min_split_improvement = 0.0001,
+        histogram_type = 'QuantilesGlobal',
+        max_abs_leafnode_pred = 1.7976931348623157e+308,
+        pred_noise_bandwidth = 0.0,
+        calibrate_model = False,
+        check_constant_response = True,
+        #      evaluate_auto = False#True
+    )
+    gbm1.train(x=myX, y="IsDepDelayed", training_frame=air_hex)
+
+    gbm2 = H2OGradientBoostingEstimator(
+        nfolds = 0,
+        keep_cross_validation_models = True,
+        keep_cross_validation_predictions = False,
+        keep_cross_validation_fold_assignment = False,
+        score_each_iteration = False,
+        score_tree_interval = 1,
+        # response_column = 'isDepDelayed',
+        #ignored_columns = ignoredCols,
+        ignore_const_cols = True,
+        balance_classes = False,
+        max_after_balance_size = 5.0,
+        max_confusion_matrix_size = 20,
+        max_hit_ratio_k = 0,
+        ntrees = 50,
+        max_depth = 10,
+        min_rows = 2.0,
+        nbins = 16,
+        nbins_top_level = 1024,
+        nbins_cats = 64,
+        r2_stopping = 1.7976931348623157e+308,
+        stopping_rounds = 0,
+        stopping_tolerance = 0.001,
+        max_runtime_secs = 1.7976931348623157e+308,
+        #check why 1249 and not 1242
+        seed = 1249,
+        build_tree_one_node = False,
+        learn_rate = 0.1,
+        learn_rate_annealing = 1.0,
+        distribution = 'bernoulli',
+        quantile_alpha = 0.5,
+        tweedie_power = 1.5,
+        huber_alpha = 0.9,
+        sample_rate = 0.75,
+        col_sample_rate = 0.31,
+        col_sample_rate_change_per_level = 0.94,
+        col_sample_rate_per_tree = 0.65,
+        min_split_improvement = 0.0001,
+        histogram_type = 'QuantilesGlobal',
+        max_abs_leafnode_pred = 1.7976931348623157e+308,
+        pred_noise_bandwidth = 0.0,
+        calibrate_model = False,
+        check_constant_response = True,
+        # evaluate_auto = False#True
+    )
+    gbm2.train(x=myX, y="IsDepDelayed", training_frame=air_hex)
+
+    rmse1 = extract_from_twoDimTable(gbm1._model_json["output"]["scoring_history"], "training_rmse", False)
+    rmse2 = extract_from_twoDimTable(gbm2._model_json["output"]["scoring_history"], "training_rmse", False)
+    print(rmse1)
+    print(rmse2)
+
+    assert equal_two_arrays(rmse1, rmse2, 1e-5, 1e-6, False), \
+        "Training_rmse are different between the two grid search models.  Tests are supposed to be repeatable in " \
+        "this case.  Make sure model seeds are actually set correctly in the Java backend."
+
+def equal_two_arrays(array1, array2, eps, tolerance, throwError=True):
```

Comment:
shoud we move this to utils?

Reply:
It is copied from pyunit_utils. I did not catch this when I copied this test from the JIRA ticket attachments... Fixed.